### PR TITLE
feat: 3pqe3k8 - eth sushi farm v2

### PR DESCRIPTION
**Pull request by 0xmainnov**

### CODE REVIEW

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+        rewardToken1 = _rewardToken1;
+        strategy = _strategy;
+        router = _router;
+        swapRouter = _swapRouter;
+        name = _name;
+        weth = _weth;
+
+        isReward = true;
+    }
+
```
Comment (daiki-iu) [inline v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:66]: @0xmainnov  no comment for deposit function here. Would you like to document this function?

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+            (success, ) = payable(_account).call{value: amountOut - taxAmount}(
+                ""
+            );
+            require(success, "Failed to send ether");
+        }
+
+        return amountOut;
+    }
+
+    function claim(uint256 _tokenId, address _account)
```
Comment (daiki-iu) [inline v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:268]: @0xmainnov  no comment for claim function here. Would you like to document this function?

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+            require(success, "Failed to send ether");
+        }
+
+        IHedgepieAdapterInfoEth(IHedgepieInvestorEth(investor).adapterInfo())
+            .updateProfitInfo(_tokenId, amountOut, true);
+
+        return amountOut;
+    }
+
+    function pendingReward(uint256 _tokenId, address _account)
```
Comment (daiki-iu) [inline v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:328]: @0xmainnov  no comment for pendingReward function here. Would you like to document this function?

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+            .updateTVLInfo(_tokenId, _amountIn, true);
+        IHedgepieAdapterInfoEth(IHedgepieInvestorEth(investor).adapterInfo())
+            .updateTradedInfo(_tokenId, _amountIn, true);
+        IHedgepieAdapterInfoEth(IHedgepieInvestorEth(investor).adapterInfo())
+            .updateParticipantInfo(_tokenId, _account, true);
+
+        return _amountIn;
+    }
+
+    function withdraw(uint256 _tokenId, address _account)
```
Comment (daiki-iu) [inline v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:139]: @0xmainnov  no comment for withdraw function here. Would you like to document this function?

File: v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+    // Deploy Pancakeswap LP Adapter contract
+    const Lib = await ethers.getContractFactory("HedgepieLibraryEth");
+    const lib = await Lib.deploy();
+    const SushiLPAdapter = await ethers.getContractFactory("SushiFarmV2AdapterEth", {
+      libraries: {
+        HedgepieLibraryEth: lib.address,
+      },
+    });
+
+    this.aAdapter = await SushiLPAdapter.deploy(
```
Comment (daiki-iu) [inline v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js:69]: @0xmainnov  aAdapter does not make sense. Would you like to rename is to "adapter" or "sushiAdapter"?

File: v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+    await this.aAdapter.setPath(this.weth, this.radar, [this.weth, this.radar]);
+    await this.aAdapter.setPath(this.radar, this.weth, [this.radar, this.weth]);
+
+    console.log("Owner: ", this.owner.address);
+    console.log("Investor: ", this.investor.address);
+    console.log("Strategy: ", strategy);
+    console.log("Info: ", this.adapterInfo.address);
+    console.log("SushiSwapFarmLPAdapter: ", this.aAdapter.address);
+
+    this.lpContract = await ethers.getContractAt("VBep20Interface", lpToken);
```
Comment (daiki-iu) [inline v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js:132]: @0xmainnov  move this line to #59

**Review by daiki-iu — CHANGES_REQUESTED** [review]
@0xmainnov left some comments.

File: v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js (hunk 6 above)
Comment (0xmainnov) [reply v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js]: Fixed

File: v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js (hunk 5 above)
Comment (0xmainnov) [reply v1-contract/test/integration/adapters/eth/sushi/sushi-farm-v2-adapter.test.js]: Fixed

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol (hunk 4 above)
Comment (0xmainnov) [reply v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:150]: Fixed

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol (hunk 3 above)
Comment (0xmainnov) [reply v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:349]: Fixed

File: v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol (hunk 2 above)
Comment (0xmainnov) [reply v1-contract/contracts/adapters/eth/sushiswap/SushiFarmV2AdapterEth.sol:284]: Fixed

**Review by daiki-iu — APPROVED** [review]
@0xmainnov  LGTM!

**Pull request merged**